core/controllers/contributor_dashboard_test.py

```diff
@@ -1785,6 +1785,207 @@ def test_get_translatable_topic_names(self) -> None:
+            response,
+            {'topic_names_per_classRoom': [
+                {
+                    'classRoom': 'Class 1',
+                    'topics': ['topic 1']
+                }
+            ]}
+        )
+
+    def test_topic_without_classroom_should_also_be_returned(
+        self) -> None:
+        # Create topics.
+        subtopic = topic_domain.Subtopic(
+            1, 'Title', ['skill_id_3'], 'image.svg',
+            constants.ALLOWED_THUMBNAIL_BG_COLORS['subtopic'][0], 21131,
+            'dummy-subtopic-three')
+        topic_id_1 = topic_fetchers.get_new_topic_id()
+        self.save_new_topic(
+            topic_id_1,
+            self.admin_id,
+            name='topic 1',
+            abbreviated_name='abbrev-one',
+            url_fragment='topic-a',
+            description='description',
+            canonical_story_ids=[],
+            additional_story_ids=[],
+            uncategorized_skill_ids=[],
+            subtopics=[subtopic],
+            next_subtopic_id=2
+        )
+        topic_id_2 = topic_fetchers.get_new_topic_id()
+        self.save_new_topic(
+            topic_id_2,
+            self.admin_id,
+            name='topic 2',
+            abbreviated_name='abbrev-two',
+            url_fragment='topic-b',
+            description='description',
+            canonical_story_ids=[],
+            additional_story_ids=[],
+            uncategorized_skill_ids=[],
+            subtopics=[subtopic],
+            next_subtopic_id=2
+        )
+        topic_id_3 = topic_fetchers.get_new_topic_id()
+        self.save_new_topic(
+            topic_id_3,
+            self.admin_id,
+            name='topic 3',
+            abbreviated_name='abbrev-three',
+            url_fragment='topic-c',
+            description='description',
+            canonical_story_ids=[],
+            additional_story_ids=[],
+            uncategorized_skill_ids=[],
+            subtopics=[subtopic],
+            next_subtopic_id=2
+        )
+
+        # Create classrooms.
+        classroom_id_1 = classroom_config_services.get_new_classroom_id()
+        self.save_new_valid_classroom(
+            classroom_id=classroom_id_1,
+            name='Class 1',
+            topic_id_to_prerequisite_topic_ids={
+                topic_id_1: []
+            }
+        )
+        classroom_id_2 = classroom_config_services.get_new_classroom_id()
+        self.save_new_valid_classroom(
+            classroom_id=classroom_id_2,
+            name='Class 2',
+            topic_id_to_prerequisite_topic_ids={
+                topic_id_2: []
+            }
+        )
+
+        # Publish the topics.
+        topic_services.publish_topic(topic_id_1, self.admin_id)
+        topic_services.publish_topic(topic_id_2, self.admin_id)
+        topic_services.publish_topic(topic_id_3, self.admin_id)
+
+        response = self.get_json('/gettranslatabletopicnamesperclassroom')
+        expected_response = {
+            'topic_names_per_classRoom': [
```

Comment:
Can we use snake_case to maintain uniformity throughout the codebase?
Ditto elsewhere